### PR TITLE
consolidate soci_index.go with index_builder.go

### PR DESCRIPTION
**Pull request by dvnguyen-amzn**
*Issue #, if available:* 
https://github.com/awslabs/soci-snapshotter/issues/128

*Description of changes:*
This change moves IndexBuilder module to soci_index.go file, and refactor the CLI and tests to use the newly added IndexBuilder module.

*Testing performed:*
`make && make test && make check`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Co-authored with: Qingshen Huang <huangqs@amazon.com>

### CODE REVIEW

File: soci/soci_index.go [hunk 1]
```diff
@@ -50,6 +50,12 @@ const (
 	IndexAnnotationBuildToolIdentifier = "com.amazon.soci.build-tool-identifier"
 	// media type for OCI Artifact manifest
 	OCIArtifactManifestMediaType = "application/vnd.oci.artifact.manifest.v1+json"
+	// default span size
```
Comment (vkuzniet) [inline soci/soci_index.go:53]: can you please also comment what the value is, e.g. 4MiB?

File: soci/soci_index.go [hunk 2]
```diff
@@ -50,6 +50,12 @@ const (
...
+	defaultSpanSize = int64(1 << 22)
+	// min layer size
```
Comment (vkuzniet) [inline soci/soci_index.go:55]: can you please also comment what the value is, e.g. 10MiB?

Comment (dvnguyen-amzn) [reply soci/soci_index.go]: done

File: soci/soci_index.go (hunk 1 above)
Comment (dvnguyen-amzn) [reply soci/soci_index.go]: done

File: soci/soci_index_test.go [hunk 3]
```diff
@@ -142,10 +142,10 @@ func TestBuildSociIndexNotLayer(t *testing.T) {
 				MediaType: tc.mediaType,
 				Digest:    "layerdigest",
 			}
-			cfg := &buildConfig{}
 			spanSize := int64(65535)
 			blobStore := memory.New()
-			_, err := buildSociLayer(ctx, cs, desc, spanSize, blobStore, cfg)
+			builder, _ := NewIndexBuilder(cs, blobStore, WithSpanSize(spanSize), WithMinLayerSize(0))
```
Comment (hanyuel) [inline soci/soci_index_test.go:147]: We only need one ``IndexBuilder`` for all test cases, so ``NewIndexBuilder`` should be moved out of this ``for`` loop.

**Review by vkuzniet — APPROVED** [review]

**Review by rdpsin — APPROVED** [review]

File: soci/soci_index.go [hunk 4]
```diff
@@ -203,7 +239,7 @@ func BuildSociIndex(ctx context.Context, cs content.Store, img images.Image, spa
 	index := NewIndex(ztocsDesc, refers, annotations)
 	return &IndexWithMetadata{
 		Index:       index,
-		Platform:    &config.platform,
+		Platform:    &b.config.platform,
```
[the remark this answers is not in the document]
Comment (vkuzniet) [reply soci/soci_index.go:242]: This will be done in another PR.

**Pull request merged**